mathics/builtin/image.py

```diff
@@ -0,0 +1,736 @@
+        pixels = skimage.img_as_ubyte(image.color_convert('RGB').pixels)
+        im = PIL.Image.fromarray(pixels).quantize(n.to_python())
+        im = im.convert('RGB')
+        return Image(numpy.array(im), 'RGB')
+
+
+class Threshold(Builtin):
+    options = {
+        'Method': '"Cluster"'
+    }
+
+    messages = {
+        'illegalmethod': 'Method `` is not supported.'
+    }
+
+    def apply(self, image, evaluation, options):
+        'Threshold[image_Image, OptionsPattern[Threshold]]'
+        pixels = image.grayscale().pixels
+
+        method = self.get_option(options, 'Method', evaluation)
+        method_name = method.get_string_value() if isinstance(method, String) else method.to_python()
+        if method_name == 'Cluster':
+            threshold = skimage.filters.threshold_otsu(pixels)
+        elif method_name == 'Median':
+            threshold = numpy.median(pixels)
+        elif method_name == 'Mean':
+            threshold = numpy.mean(pixels)
+        else:
+           return evaluation.error('Threshold', 'illegalmethod', method)
+
+        return Real(threshold)
+
+
+class Binarize(Builtin):
+    def apply(self, image, evaluation):
+        'Binarize[image_Image]'
+        image = image.grayscale()
+        threshold = Expression('Threshold', image).evaluate(evaluation).to_python()
+        return Image(image.pixels > threshold, 'Grayscale')
+
+    def apply_t(self, image, t, evaluation):
+        'Binarize[image_Image, t_?RealNumberQ]'
+        pixels = image.grayscale().pixels
+        return Image(pixels > t.to_python(), 'Grayscale')
+
+    def apply_t1_t2(self, image, t1, t2, evaluation):
+        'Binarize[image_Image, {t1_?RealNumberQ, t2_?RealNumberQ}]'
+        pixels = image.grayscale().pixels
+        mask1 = pixels > t1.to_python()
+        mask2 = pixels < t2.to_python()
+        return Image(mask1 * mask2, 'Grayscale')
+
+
+class ColorNegate(Builtin):
+    def apply(self, image, evaluation):
+        'ColorNegate[image_Image]'
+        pixels = image.pixels
+        anchor = numpy.ndarray(pixels.shape, dtype=pixels.dtype)
+        anchor.fill(skimage.dtype_limits(pixels)[1])
+        return Image(anchor - pixels, image.color_space)
+
+
+class ColorSeparate(Builtin):
+    def apply(self, image, evaluation):
+        'ColorSeparate[image_Image]'
+        images = []
+        pixels = image.pixels
+        if len(pixels.shape) < 3:
+            images.append(pixels)
+        else:
+            for i in range(pixels.shape[2]):
+                images.append(Image(pixels[:, :, i], 'Grayscale'))
+        return Expression('List', *images)
+
+
+class Colorize(Builtin):
+    messages = {
+        'toomany': 'Too many levels.'
+    }
+
+    def apply(self, a, evaluation):
+        'Colorize[a_?MatrixQ]'
+
+        a = numpy.array(a.to_python())
+        n = int(numpy.max(a)) + 1
+        if n > 8192:
+            return evaluation.error('Colorize', 'toomany')
+
+        cmap = matplotlib.cm.get_cmap('hot', n)
```

Comment:
It might be worth splitting the matplotlib dependency out (separate to scikit-image) if this is the only place it's used.